sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -346,5 +346,20 @@ public static bool IsLeftSideOfAssignment(this ExpressionSyntax expression)
                 topParenthesizedExpression.Parent is AssignmentExpressionSyntax assignment &&
                 assignment.Left == topParenthesizedExpression;
         }
+
+        public static bool IsComment(this SyntaxTrivia trivia)

```

Comment:
This code was duplicated by the Fixme rule, the Todo rule and the comment metrics analyzers.